conf/lsf.conf

```
@@ -32,41 +33,46 @@ backend {
         runtime-attributes = """
         Int cpu = 1
         Int hosts = 1
-        Float? memory_mb = 4096
-        String lsf_queue = "standard"
+        Int? memory_mb = 4096
+        String lsf_queue = "compbio"
         String? lsf_job_group
-        String? docker 
+        String? docker
+        # Float memory_per_cpu = memory_mb / cpu
         """
 
+        #submit = """bsub -J ${job_name} -cwd ${cwd} -o ${out} -e ${err} -R rusage[mem=${memory_mb + "MB"}] -R "span[hosts=1]" -n ${cpu} ${"-q " + lsf_queue}  /usr/bin/env bash ${script}"""
+
         submit = """
           bsub \
-            -q ~{lsf_queue} \
-            -n ~{cpu} \
-            ~{"-g " + lsf_job_group} \
-            -R "rusage[mem=~{round(round(memory_mb)/cpu)}] span[hosts=~{hosts}]" \
-            -J ~{job_name} \
-            -cwd ~{cwd} \
-            -o ~{out} \
-            -e ~{err} \
-            /usr/bin/env bash ~{script}
+            -q ${lsf_queue} \
+            -n ${cpu} \
+            ${"-g " + lsf_job_group} \
+            -R "rusage[mem=${round(round(memory_mb)/cpu)}] span[hosts=${hosts}]" \
+            -J ${job_name} \
+            -cwd ${cwd} \
+            -o ${cwd}/execution/stdout.lsf \
+            -e ${cwd}/execution/stderr.lsf \
+            /usr/bin/env bash ${script}
         """
 
         submit-docker = """
           bsub \
-            -q ~{lsf_queue} \
-            -n ~{cpu} \
-            ~{"-g " + lsf_job_group} \
-            -R "rusage[mem=~{round(round(memory_mb)/cpu)}] span[hosts=~{hosts}]" \
-            -J ~{job_name} \
-            -cwd ~{cwd} \
-            -o ~{cwd}/execution/stdout.lsf \
-            -e ~{cwd}/execution/stderr.lsf \
-            "singularity exec --bind ~{cwd}:~{docker_cwd} docker://~{docker} ~{job_shell} ~{script}"
+            -q ${lsf_queue} \
+            -n ${cpu} \
+            ${"-g " + lsf_job_group} \
+            -R "rusage[mem=${round(round(memory_mb)/cpu)}] span[hosts=${hosts}]" \
+            -J ${job_name} \
+            -cwd ${cwd} \
+            -o ${cwd}/execution/stdout.lsf \
+            -e ${cwd}/execution/stderr.lsf \
+            "singularity exec --bind ${cwd}:${docker_cwd} docker://${docker} ${job_shell} ${script}"
         """
 
 
-        kill = "bkill ~{job_id}"
-        check-alive = "bjobs ~{job_id}"
+        kill = "bkill ${job_id}"
+        check-alive = "bjobs ${job_id}"
+        # check-alive = "check-job-alive ${job_id}"
```

```suggestion
        check-alive = "check-job-alive ${job_id}"
```

Maybe we need to make a comment that the `check-job-alive` script must be in the user's path.

---

Yes that needs to be documented somewhere. Right now `check-job-alive` is the only file in `bin/` and it's purpose is very unclear.

---

For internal usage, we have https://wiki.stjude.org/display/devops/WDL+integration+with+LSF where I intend to add documentation about `check-job-alive`. For external users, do you have a preference on where to document that? An inline comment seems easy to overlook.

---

I'm not sure where it would be best for external users right now. It could go in the README, but it's very out of date at the moment and there's nowhere in the current doc it makes sense to put information about LSF configuration. Updating the whole thing needs to be done at some point, and when that's done it would make sense to include a section on configuration.

---

I think for now an inline comment is okay but I agree it will probably be overlooked. If they don't have `check-job-alive`, I believe cromwell will fail pretty early in the run so that should hopefully guide people to looking for some documentation about it. We should probably make updates to the README later and/or make Github wiki pages if there's a lot of things we want to document.

---

I inlined the comment, but also added bin as a section in the README.

---

I think this is good till we update the README/make a wiki 👍